ffi/src/scan.rs

```diff
@@ -56,19 +65,20 @@ pub struct ArrowFFIData {
 /// data_handle must be a valid EngineDataHandle as read by the
 /// [`delta_kernel::engine::default::DefaultEngine`] obtained from `get_default_engine`.
 #[cfg(feature = "default-engine")]
+#[no_mangle]
 pub unsafe extern "C" fn get_raw_arrow_data(
-    data_handle: *mut EngineDataHandle,
-    engine: *const ExternEngineHandle,
+    data: Handle<EngineDataHandle>,
+    engine: Handle<SharedExternEngine>,
 ) -> ExternResult<*mut ArrowFFIData> {
-    get_raw_arrow_data_impl(data_handle).into_extern_result(engine)
+    get_raw_arrow_data_impl(data).into_extern_result(engine)
 }
 
 #[cfg(feature = "default-engine")]
 unsafe fn get_raw_arrow_data_impl(
-    data_handle: *mut EngineDataHandle,
+    data: Handle<EngineDataHandle>,
 ) -> DeltaResult<*mut ArrowFFIData> {
-    let boxed_data = unsafe { Box::from_raw(data_handle) };
-    let data = boxed_data.data;
+    // TODO(frj): This consumes the handle. Is that what we really want?
```

Comment:
I believe so yet. afaict this transfers ownership of the data to the engine. It's a little unclear from the arrow docs, but if I try and free the data after using this I get a segfault so I'm pretty sure

Reply:
Yes, today the code transfers ownership to the engine... but _should_ it? Do we _want_ to be allocating that data over and over? Or allocating once and rewriting it over and over?

Reply:
Well, in this case we want to transfer actual raw arrow data over. So either we have to `clone` it and keep the previous one alive, or consume it. 

From my comments in the code below , I wasn't quite clear if `let array = arrow_data::ffi::FFI_ArrowArray::new(&array_data);` consumed the data, but afaict, it does.

So for efficiency's sake I think we'd want to keep the semantics that this destroys passed in data and you get back the raw arrow version of it.